drivers/include/periph/gpio.h

```diff
@@ -21,239 +21,150 @@
+ * @brief   Definition of possible active flanks for external interrupt mode
  */
+#ifndef OVRD_GPIO_FLANK_T
 typedef enum {
     GPIO_FALLING = 0,       /**< emit interrupt on falling flank */
     GPIO_RISING = 1,        /**< emit interrupt on rising flank */
     GPIO_BOTH = 2           /**< emit interrupt on both flanks */
 } gpio_flank_t;
+#endif
 
 /**
- * @brief Signature for function called from interrupt context after an external event
+ * @brief   Signature of event callback functions triggered from interrupts
  *
  * @param[in] arg       optional context for the callback
  */
 typedef void (*gpio_cb_t)(void *arg);
 
 /**
- * @brief Initialize GPIO pin as output pin
+ * @brief   Initialize the given pin as general purpose input or output
  *
- * @param[in] dev       the GPIO pin to set as output
- * @param[in] pullup    define if pull resistors should be used
+ * @param[in] pin       pin to initialize
+ * @param[in] dir       direction for the pin, input or output
+ * @param[in] pullup    define what pull resistors should be used
  *
  * @return              0 on success
- * @return              -1 on pull-up mode unavailable
+ * @return              -1 on error
  */
-int gpio_init_out(gpio_t dev, gpio_pp_t pullup);
+int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup);
 
 /**
- * @brief Initialize a GPIO pin in input mode
+ * @brief   Initialize a GPIO pin for external interrupt usage
  *
- * @param[in] dev       the GPIO pin to use as input
- * @param[in] pullup    define if pull resistors should be used
- *
- * @return              0 on success
- * @return              -1 on pull-up mode unavailable
- */
-int gpio_init_in(gpio_t dev, gpio_pp_t pullup);
-
-/**
- * @brief Initialize a GPIO pin for external interrupt usage
- *
- * The registered callback function will be called in interrupt context every time the defined flank(s)
- * are detected.
+ * The registered callback function will be called in interrupt context every
+ * time the defined flank(s) are detected.
  *
  * The interrupt is activated automatically after the initialization.
  *
- * @param[in] dev       the GPIO pin to initialize
- * @param[in] pullup    define if pull resistors should be enabled
+ * @param[in] pin       pin to initialize
+ * @param[in] pullup    define what pull resistors should be enabled
  * @param[in] flank     define the active flank(s)
- * @param[in] cb        register the callback that is called in interrupt context
+ * @param[in] cb        callback that is called from interrupt context
  * @param[in] arg       optional argument passed to the callback
  *
  * @return              0 on success
- * @return              -1 on pull-up mode unavailable
- * @return              -2 on active flank unavailable
+ * @return              -1 on error
  */
-int gpio_init_int(gpio_t dev, gpio_pp_t pullup, gpio_flank_t flank, gpio_cb_t cb, void *arg);
+int gpio_init_exti(gpio_t pin, gpio_pp_t pullup, gpio_flank_t flank,
```

Comment:
not really. As I remember that discussion, the conclusion was, that this change is sensible bu is kind of unrelated to this PR. But as we are touching all this GPIO stuff here anyway, we leave it in.